scripts/suite_objects.py

```diff
@@ -1184,6 +1192,31 @@ def analyze(self, phase, group, scheme_library, suite_vars, level):
                                                   vstdname))
                 # end if
             # end if
+            # Are there any forward/reverse transforms for this variable?
+            if compat_obj is not None and (compat_obj.has_vert_transforms or compat_obj.has_unit_transforms):
+                # Add local variable (<var>_local) needed for transformation.
+                tmp_var = var.clone(var.get_prop_value('local_name')+'_local')
+                self.__group.manage_variable(tmp_var)
+
+                # Create indices, flipping if necessary.
+                indices = [':']*var.get_rank()
+                if compat_obj.has_vert_transforms:
+                    dim = find_vertical_dimension(var.get_dimensions())
+                    lvar = 'nlev'
```

Comment:
@gold2718 @climbfuji @peverwhee 
How do we populate "lvar" with the correct local_name, "nlev", for the standard_name "vertical_layer_dimension"?
(Please someone help me out with the magic needed to do this!!!!)

As far as I can tell, we are in the variable loop within analyze, and we don't have access to the other variables(?), just the current one, so my workaround was to create a local variable using the current variable in the loop, local_array_nlay = size(local_array,dim). But this is not ideal from our conversations (I'm still not convinced why, but anyways...)

Reply:
Sorry for the delay, busy day . . .
Inside the block `if compat_obj is not None and (compat_obj.has_vert_transforms or compat_obj.has_unit_transforms):', I tried:
```
                vdim_name = vert_dim.split(':')[-1]
                group_vvar = self.__group.call_list.find_variable(vdim_name)
                vname = vvar.get_prop_value('local_name')
```
which gets me `nlev`. Is this what you were looking for?

Reply:
No worries at all. That piece did the trick, nice and slick. Thanks!
![Screenshot 2023-10-31 at 1 48 45 PM](https://github.com/NCAR/ccpp-framework/assets/13949033/c16f9c59-b20a-4d81-9d89-2c5d725a29e0)